### ISSUE TITLE
Add filtering or sub-search feature to Find References panel

### ISSUE THREAD

**Issue by macraig** [sourcegraph/sourcegraph#28480]
#### Feature request description

Find references has a hard limit for displayed results, the limit should be removed, filtering and search functionality should be added to the Find References panel to allow for relevant reference filtering. 

#### Is your feature request related to a problem? If so, please describe.

Find references has a hard 500 limit for displayed results. This means that a user might get 500+ precise references but can't access all of them. Even if they could, it's almost impossible to find relevant references in a large list.

#### Additional context
Feature requested by customer, design and scope TBD.

**Comment by shuyangfang** [comment]
We think to start with, removing the hard 500 result limit and allowing configuration on our end would be helpful as well. The filtering / sub-search functionality is definitely the desired end goal for us, but in the meantime, we would derive value just from being able to see references beyond the current limit. Thanks!

**Comment by macraig** [comment]
@shuyangfang thanks for your feedback! We'll start working on allowing configuration as a first step.

**Comment by macraig** [comment]
@shuyangfang FYI https://github.com/sourcegraph/sourcegraph/pull/29629 is now merged and will be available in our 3.36 release.

**Comment by jjinnii** [comment]
@macraig @mrnugget FYI I changed the name of this issue to "support large lists of references". It seems like the goal for this issue is to remove the 500 hard limit and have a design that supports being able to view a large list of references (which should include scrolling/pagination considerations), with filtering being a closely-related desired feature. 

I've captured the feature request for filtering/sub search in new issue #31955

**Comment by mrnugget** [comment]
> It seems like the goal for this issue is to remove the 500 hard limit and have a design that supports being able to view a large list of references (which should include scrolling/pagination considerations), with filtering being a closely-related desired feature.

Just FYI: the 500 hard limit has been removed (see https://github.com/sourcegraph/sourcegraph/issues/28480#issuecomment-1012278957) and we do have infinite pagination.

Besides general UX concerns around "how do we make a list with 1000 elements easy to navigate?" I think what's left of this ticket here is only the filtering.

**Comment by jjinnii** [comment]
Thanks @mrnugget. Then this issue #30971 was implementing pagination specific to the PoC then? 

And as you mentioned, I think there's a lot of UX thinking to be done around navigating a massive list of results. Would it be helpful or more confusing if I closed out this issue and created a new one specific to the user problemsof scrolling through a large list of results with the currently implemented design? 

Just trying to make sure it's crystal clear what feature and user problem each issue in the tracking issue is intended to cover.

**Comment by mrnugget** [comment]
> Then this issue #30971 was implementing pagination specific to the PoC then?

Yup. It's also part of the tracking issue for the experimental reference panel: https://github.com/sourcegraph/sourcegraph/issues/30227 The non-experimental reference panel has had pagination for a long time.

> Would it be helpful or more confusing if I closed out this issue and created a new one specific to the user problemsof scrolling through a large list of results with the currently implemented design?

It depends, I guess. Do you mean "filtering" with "user problems of scrolling [...]"? Because this ticket right here was/is about two things:

1. Removing hard limit of 500
2. Allowing filtering of lists (see this comment above: https://github.com/sourcegraph/sourcegraph/issues/28480#issuecomment-1001662111)

Hard limit is removed, so what's left is allowing filtering.
